updater/aws_test.go

```diff
@@ -15,6 +15,61 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetCommandResult(t *testing.T) {
+	cases := []struct {
+		name          string
+		invocationOut *ssm.GetCommandInvocationOutput
+		expectedError string
+		expectedOut   []byte
+		getError      error
+	}{
+		{
+			name: "getCommand success",
+			invocationOut: &ssm.GetCommandInvocationOutput{
+				Status:                aws.String("Success"),
+				StandardOutputContent: aws.String("OutputContent"),
+			},
+			expectedOut: []byte(aws.StringValue(aws.String("OutputContent"))),
+			getError:    nil,
+		},
+		{
+			name:          "getCommand fail",
+			getError:      errors.New("failed to get command invocation"),
+			expectedError: "failed to retrieve command invocation output: failed to get command invocation",
+			invocationOut: nil,
+			expectedOut:   nil,
+		},
+		{
+			name: "command status non-Success",
+			invocationOut: &ssm.GetCommandInvocationOutput{
+				Status:                aws.String("TimedOut"),
+				StandardOutputContent: nil,
+			},
+			expectedError: "command command-id has not reached success status, current status \"TimedOut\"",
+			expectedOut:   nil,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			mockSSM := MockSSM{
+				GetCommandInvocationFn: func(input *ssm.GetCommandInvocationInput) (*ssm.GetCommandInvocationOutput, error) {
+					assert.Equal(t, "command-id", aws.StringValue(input.CommandId))
+					assert.Equal(t, "instance-id", aws.StringValue(input.InstanceId))
+					return tc.invocationOut, tc.getError
+				},
+			}
+			u := updater{ssm: mockSSM}
+			actual, err := u.getCommandResult("command-id", "instance-id")
+			if tc.expectedOut != nil {
+				require.NoError(t, err)
+				assert.EqualValues(t, tc.expectedOut, actual)
+			} else {
```

Comment:
require.Error(t, err) is important too..to check if error occured before checking error equality.

Reply:
Understood. I had thought that if no error occurred then the errors would not be equal and result in a test failure. Thanks for pointing it out.